File: lib/cfr/builtins/libs/flutter/painting/borderRadius.dart
import 'dart:core' as _fac9;
import 'dart:ui' as _a643;

import 'package:flutter/src/painting/border_radius.dart' as _9d8a;

import 'package:hydro_sdk/cfr/runtimeSupport.dart' as _36c2;

class VMManagedBorderRadius extends _36c2.VMManagedBox<_9d8a.BorderRadius> {
  VMManagedBorderRadius(
      {required this.table, required this.vmObject, required this.hydroState})
      : super(
          table: table,
          vmObject: vmObject,
          hydroState: hydroState,
        ) {
    table['topLeft'] = _36c2.maybeBoxObject<_a643.Radius>(
        object: vmObject.topLeft,
        hydroState: hydroState,
        table: _36c2.HydroTable());
    table['topRight'] = _36c2.maybeBoxObject<_a643.Radius>(
        object: vmObject.topRight,
        hydroState: hydroState,
        table: _36c2.HydroTable());
    table['bottomLeft'] = _36c2.maybeBoxObject<_a643.Radius>(
        object: vmObject.bottomLeft,
        hydroState: hydroState,
        table: _36c2.HydroTable());
    table['bottomRight'] = _36c2.maybeBoxObject<_a643.Radius>(
        object: vmObject.bottomRight,
        hydroState: hydroState,
        table: _36c2.HydroTable());
    table['copyWith'] = _36c2.makeLuaDartFunc(
        func: (_fac9.List<_fac9.dynamic> luaCallerArguments) {
      return [
        _36c2.maybeBoxObject<_9d8a.BorderRadius>(
            object: vmObject.copyWith(
                bottomLeft:
                    _36c2.maybeUnBoxAndBuildArgument<_a643.Radius?, _fac9.dynamic>(
                        luaCallerArguments.length >= 2
                            ? luaCallerArguments[1]['bottomLeft']
                            : null,
                        parentState: hydroState),
                bottomRight:
                    _36c2.maybeUnBoxAndBuildArgument<_a643.Radius?, _fac9.dynamic>(luaCallerArguments.length >= 2 ? luaCallerArguments[1]['bottomRight'] : null,
                        parentState: hydroState),
                topLeft: _36c2.maybeUnBoxAndBuildArgument<_a643.Radius?, _fac9.dynamic>(
                    luaCallerArguments.length >= 2 ? luaCallerArguments[1]['topLeft'] : null,
                    parentState: hydroState),
                topRight: _36c2.maybeUnBoxAndBuildArgument<_a643.Radius?, _fac9.dynamic>(luaCallerArguments.length >= 2 ? luaCallerArguments[1]['topRight'] : null, parentState: hydroState)),
            hydroState: hydroState,
            table: _36c2.HydroTable()),
      ];
    });
    table['toRRect'] = _36c2.makeLuaDartFunc(
        func: (_fac9.List<_fac9.dynamic> luaCallerArguments) {
      return [
        _36c2.maybeBoxObject<_a643.RRect>(
            object: vmObject.toRRect(
                _36c2.maybeUnBoxAndBuildArgument<_a643.Rect, _fac9.dynamic>(
                    luaCallerArguments[1],
                    parentState: hydroState)),
            hydroState: hydroState,
            table: _36c2.HydroTable()),
      ];
    });
    table['subtract'] = _36c2.makeLuaDartFunc(
        func: (_fac9.List<_fac9.dynamic> luaCallerArguments) {
      return [
        _36c2.maybeBoxObject<_9d8a.BorderRadiusGeometry>(
            object: vmObject.subtract(_36c2.maybeUnBoxAndBuildArgument<
                _9d8a.BorderRadiusGeometry,
                _fac9.dynamic>(luaCallerArguments[1], parentState: hydroState)),
            hydroState: hydroState,
            table: _36c2.HydroTable()),
      ];
    });
    table['add'] = _36c2.makeLuaDartFunc(
        func: (_fac9.List<_fac9.dynamic> luaCallerArguments) {
      return [
        _36c2.maybeBoxObject<_9d8a.BorderRadiusGeometry>(
            object: vmObject.add(_36c2.maybeUnBoxAndBuildArgument<
                _9d8a.BorderRadiusGeometry,
                _fac9.dynamic>(luaCallerArguments[1], parentState: hydroState)),
            hydroState: hydroState,
            table: _36c2.HydroTable()),
      ];
    });
    table['resolve'] = _36c2.makeLuaDartFunc(
        func: (_fac9.List<_fac9.dynamic> luaCallerArguments) {
      return [
        _36c2.maybeBoxObject<_9d8a.BorderRadius>(
            object: vmObject.resolve(_36c2.maybeUnBoxEnum(
                values: _a643.TextDirection.values,
                boxedEnum: luaCallerArguments[1])),
            hydroState: hydroState,
            table: _36c2.HydroTable()),
      ];
    });
    table['toString'] = _36c2.makeLuaDartFunc(
        func: (_fac9.List<_fac9.dynamic> luaCallerArguments) {
      return [
        vmObject.toString(),
      ];
    });
    table['getHashCode'] = _36c2.makeLuaDartFunc(
        func: (_fac9.List<_fac9.dynamic> luaCallerArguments) {
      return [
        vmObject.hashCode,
      ];
    });
  }

  final _36c2.HydroTable table;

  final _36c2.HydroState hydroState;

  final _9d8a.BorderRadius vmObject;
}

void loadBorderRadius(
    {required _36c2.HydroState hydroState, required _36c2.HydroTable table}) {
  table['borderRadiusLerp'] = _36c2.makeLuaDartFunc(
      func: (_fac9.List<_fac9.dynamic> luaCallerArguments) {
    final returnValue = _9d8a.BorderRadius.lerp(
        _36c2.maybeUnBoxAndBuildArgument<_9d8a.BorderRadius?, _fac9.dynamic>(
            luaCallerArguments[1],
            parentState: hydroState),
        _36c2.maybeUnBoxAndBuildArgument<_9d8a.BorderRadius?, _fac9.dynamic>(
            luaCallerArguments[2],
            parentState: hydroState),
        luaCallerArguments[3]?.toDouble());
    if (returnValue != null) {
      return [
        _36c2.maybeBoxObject<_9d8a.BorderRadius?>(
            object: returnValue,
            hydroState: hydroState,
            table: _36c2.HydroTable()),
      ];
    }
    return [];
  });
  _36c2.registerBoxer<_9d8a.BorderRadius>(boxer: (
      {required _9d8a.BorderRadius vmObject,
      required _36c2.HydroState hydroState,
      required _36c2.HydroTable table}) {
    return VMManagedBorderRadius(
        vmObject: vmObject, hydroState: hydroState, table: table);
  });
}

File: runtime/flutter/painting/borderRadius.ts
import { IRadius, Radius } from "../../dart/ui/radius";
import { IRect } from "../../dart/ui/rect";
import { IRRect } from "../../dart/ui/rRect";
import { TextDirection } from "../../dart/ui/textDirection";
import { IBorderRadiusGeometry } from "./borderRadiusGeometry";
declare const flutter: {
    painting: {
        borderRadiusLerp: (
            a: IBorderRadius | undefined,
            b: IBorderRadius | undefined,
            t: number
        ) => IBorderRadius | undefined;
    };
};
export interface IBorderRadius {
    topLeft: IRadius;
    topRight: IRadius;
    bottomLeft: IRadius;
    bottomRight: IRadius;
    copyWith: (props?: {
        bottomLeft?: IRadius | undefined;
        bottomRight?: IRadius | undefined;
        topLeft?: IRadius | undefined;
        topRight?: IRadius | undefined;
    }) => IBorderRadius;
    toRRect: (rect: IRect) => IRRect;
    subtract: (other: IBorderRadiusGeometry) => IBorderRadiusGeometry;
    add: (other: IBorderRadiusGeometry) => IBorderRadiusGeometry;
    resolve: (direction?: TextDirection | undefined) => IBorderRadius;
    toString: () => string;
    getHashCode: () => number;
}
export class BorderRadius implements IBorderRadiusGeometry {
    public static zero = BorderRadius.all(Radius.zero);
    public readonly topLeft: IRadius = undefined as any;
    public readonly topRight: IRadius = undefined as any;
    public readonly bottomLeft: IRadius = undefined as any;
    public readonly bottomRight: IRadius = undefined as any;
    public static lerp(
        a: IBorderRadius | undefined,
        b: IBorderRadius | undefined,
        t: number
    ): IBorderRadius | undefined {
        return flutter.painting.borderRadiusLerp(a, b, t);
    }
    private readonly _dart_copyWith: (props?: {
        bottomLeft?: IRadius | undefined;
        bottomRight?: IRadius | undefined;
        topLeft?: IRadius | undefined;
        topRight?: IRadius | undefined;
    }) => IBorderRadius = undefined as any;
    private readonly _dart_toRRect: (rect: IRect) => IRRect = undefined as any;
    private readonly _dart_subtract: (
        other: IBorderRadiusGeometry
    ) => IBorderRadiusGeometry = undefined as any;
    private readonly _dart_add: (
        other: IBorderRadiusGeometry
    ) => IBorderRadiusGeometry = undefined as any;
    private readonly _dart_resolve: (
        direction?: TextDirection | undefined
    ) => IBorderRadius = undefined as any;
    private readonly _dart_toString: () => string = undefined as any;
    private readonly _dart_getHashCode: () => number = undefined as any;
    public copyWith(props?: {
        bottomLeft?: IRadius | undefined;
        bottomRight?: IRadius | undefined;
        topLeft?: IRadius | undefined;
        topRight?: IRadius | undefined;
    }): IBorderRadius {
        return this._dart_copyWith(props);
    }
    public toRRect(rect: IRect): IRRect {
        return this._dart_toRRect(rect);
    }
    public subtract(other: IBorderRadiusGeometry): IBorderRadiusGeometry {
        return this._dart_subtract(other);
    }
    public add(other: IBorderRadiusGeometry): IBorderRadiusGeometry {
        return this._dart_add(other);
    }
    public resolve(direction?: TextDirection | undefined): IBorderRadius {
        return this._dart_resolve(direction);
    }
    public toString(): string {
        return this._dart_toString();
    }
    public getHashCode(): number {
        return this._dart_getHashCode();
    }
}
